spec/models/portfolio_spec.rb

```diff
@@ -1,14 +1,60 @@
 describe Portfolio do
   let(:portfolio)      { create(:portfolio) }
   let(:portfolio_item) { create(:portfolio_item) }
+  let(:tenant)         { create(:tenant) }
 
-  describe "#add_portfolio_item" do
-    it "returns the last added portfolio_item" do
-      expect(portfolio.add_portfolio_item(portfolio_item.id).first).to be_a PortfolioItem
+  context "without current_tenant" do
+
+    before do
+      ActsAsTenant.current_tenant = nil
     end
 
-    it "adds the portfolio_item passed into the method" do
-      expect(portfolio.add_portfolio_item(portfolio_item.id).first.id).to eq portfolio_item.id
+    describe "#add_portfolio_item" do
+      it "has no tenant set" do
+        expect(portfolio.tenant_id).to be_nil
+        expect(portfolio_item.tenant_id).to be_nil
+      end
+
+      it "returns the last added portfolio_item" do
+        expect(portfolio.add_portfolio_item(portfolio_item.id).first).to be_a PortfolioItem
+      end
+
+      it "adds the portfolio_item passed into the method" do
+        expect(portfolio.tenant_id).to be_nil
+        expect(portfolio.add_portfolio_item(portfolio_item.id).first.id).to eq portfolio_item.id
+      end
+
+      it "does not scope tenant_id on a portfolio lookup" do
+        expect(portfolio.tenant_id).to be_nil
+        expect(Portfolio.all.map(&:tenant_id)).to eq [nil]
+      end
+    end
+  end
+
+  context "with current_tenant" do
+
+    before do
+      ActsAsTenant.current_tenant = tenant
+    end
+
+    describe "#add_portfolio_item" do
+      it "has a tenant set" do
+        expect(portfolio.tenant_id).to eq tenant.id
+        expect(portfolio_item.tenant_id).to eq tenant.id
+      end
+
+      it "returns the last added portfolio_item with a tenant" do
+        expect(portfolio.add_portfolio_item(portfolio_item.id).first.tenant_id).to eq tenant.id
+      end
+
+      it "adds the portfolio_item passed into the method with a tenant" do
+        expect(portfolio.add_portfolio_item(portfolio_item.id).first.tenant_id).to eq tenant.id
+      end
+
+      it "defines a scoped tenant_id on a portfolio lookup" do
+        expect(portfolio.tenant_id).to_not be_nil
+        expect(Portfolio.all.map(&:tenant_id)).to eq [tenant.id]
```

Comment:
I was thinking more along the lines of having a block in the test that creates a Portfolio instance in another tenant, then outside of that block `Portfolio.all` should only return the current tenant object.  Here we only have 1 Portfolio object in the DB, right?